Kernel/Arch/x86/CPU.h

```diff
@@ -608,19 +614,21 @@ struct ProcessorMessageEntry {
 };
 
 struct DeferredCallEntry {
+    using HandlerFunction = Function<void()>;
+
     DeferredCallEntry* next;
-    union {
-        struct {
-            void (*handler)();
-        } callback;
-        struct {
-            void* data;
-            void (*handler)(void*);
-            void (*free)(void*);
-        } callback_with_data;
-    };
-    bool have_data;
+    alignas(HandlerFunction) u8 handler_storage[sizeof(HandlerFunction)];
     bool was_allocated;
+
+    Function<void()>& handler_value()
```

Comment:
Can't we use the using alias here as well? 
```suggestion
    HandlerFunction& handler_value()
```

Reply:
Sure.